src/hotspot/cpu/aarch64/interp_masm_aarch64.cpp

```diff
@@ -1398,6 +1399,11 @@ void InterpreterMacroAssembler::_interp_verify_oop(Register reg, TosState state,
 
 void InterpreterMacroAssembler::verify_FPU(int stack_depth, TosState state) { ; }
 
+void InterpreterMacroAssembler::trigger_action_on_enter() {
+  if (!FLAG_IS_DEFAULT(AOTCreateOnMethodEntry) && CDSPreimage == nullptr) {
```

Comment:
Does a non-nullptr value for CDSPreimage mean the AOT cache has already been created?

Reply:
its use to point at the AOT preimage from phase1, so I infer when its present that we have completed phase1 and I should not generate the triggers (as we dont want those in the cache)

Reply:
the cache is generated from phase2 onwards